### PR TITLE
Add functionality to get efi variables

### PR DESCRIPTION
**Pull request by GD-AW**
With recent activities in regards to bootkits, like [BlackLotus UEFI bootkit: Myth confirmed](https://www.welivesecurity.com/2023/03/01/blacklotus-uefi-bootkit-myth-confirmed/), I think this threat could be seen more often in future. Therefore I have implemented a possibility to analyse efi variables, to be able to identify if anything was manipulated in the system.

### CODE REVIEW

File: artifacts/definitions/Generic/System/EfiSignatures.yaml [hunk 1]
```diff
@@ -0,0 +1,104 @@
...
+      LET DBX = Select * FROM foreach(
+          row=GetSignatures(Namespace="{d719b2cb-3d3a-4596-a3bc-dad00e67656f}", Name="dbx"),
+          query={
+              Select * From foreach(
+                  row=Signatures,
+                  query={
+                      Select * FROM foreach(
+                          row=Payload.Hashes,
+                          query={
+                              Select Name, Owner, Hash as Signature From scope()
```
[the remark this answers is not in the document]
Comment (GD-AW) [reply artifacts/definitions/Generic/System/EfiSignatures.yaml]: Yes, now it is more logical

File: vql/efivariables.go [hunk 2]
```diff
@@ -0,0 +1,64 @@
+package vql
```
[the remark this answers is not in the document]
Comment (GD-AW) [reply vql/efivariables.go]: Good idea, done

File: artifacts/definitions/Generic/System/EfiSignatures.yaml [hunk 3]
```diff
@@ -0,0 +1,104 @@
+name: Generic.System.EfiSignatures
```
[the remark this answers is not in the document]
Comment (GD-AW) [reply artifacts/definitions/Generic/System/EfiSignatures.yaml:1]: I'm with you, but I don't know how to test this. The corresponding variables depend on the system it is run and AFAIK can not be modified in the OS itself, only in Bios/UEFI environment.

Do you have any good idea how one can test this?

[the remark this answers is not in the document]
Comment (scudette) [reply artifacts/definitions/Generic/System/EfiSignatures.yaml:1]: No worries. I added a test to the VQL parser. I agree it is difficult to test the OS side but it is possible to extract an EFI blob on a real OS, save it to a file then mock the plugin and test the parser.

**Review by scudette — APPROVED** [review]

**Pull request merged**